File: self_library/src/main/java/com/wjf/self_library/http/HttpResult.java
package com.wjf.self_library.http;

import android.support.annotation.NonNull;

import java.io.Serializable;

public class HttpResult<T> implements Serializable {
    private int resultCode;
    private String message;
    private T data;

    public int getResultCode() {
        return resultCode;
    }

    public void setResultCode(int resultCode) {
        this.resultCode = resultCode;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public T getData() {
        return data;
    }

    public void setData(T data) {
        this.data = data;
    }

    @NonNull
    @Override
    public String toString() {
        return "HttpResult{"+
                "data="+data+
                ", resultCode="+resultCode+
                ", message="+message+'\''+
                '}';
    }
}

File: self_library/src/main/java/com/wjf/self_library/http/HttpResult.kt
package com.wjf.self_library.http

import com.google.gson.Gson
import java.io.Serializable

/**
 * @author WJF
 */
data class HttpResult<T>(
    var resultCode: Int,
    var message: String,
    var data: T
) : Serializable {
    override fun toString(): String {
        return "resultCode：$resultCode\nmessage：$message\ndata：${Gson().toJson(data)}"
    }
}
